### PR TITLE
fix(executor): fail clearly when managed worktree disappears

### PR DESCRIPTION
**Pull request by dimileeh**
Automatically opened by AWF workspace `ws_19325816735d4d5585cb069a` (agent: `codex`).

### Task
Implement a reliability fix for the AWF executor when the managed worktree disappears mid-execution.

Observed dogfood failure:
- executor.fix_pass_add_failed logged: fatal: cannot change to '/Users/.../worktrees/<workspace_id>': No such file or directory
- AWF then kept trying validation fix passes and eventually reported a misleading validation_failure.

Expected behavior:
- Before post-agent commit, validation fix-agent runs, fix-pass git operations, and PR push, detect when the managed worktree path is missing or no longer a directory.
- Stop the executor cleanly instead of looping.
- Mark the workspace failed with infrastructure_failure and an actionable reason code/message such as WORKTREE_MISSING.
- Finish any active validation operation as failed when this happens during validation/fix-cycle.
- Preserve existing cancel/destroy stale-status behavior; do not convert operator-destroyed/cancelled workspaces into failed ones.
- Emit an event/log that makes the root cause visible to the console and API.

TDD coverage to add:
- Worktree missing before post-agent commit fails as infrastructure_failure.
- Worktree removed after validation fails but before a fix-pass git add stops immediately, no repeated fix attempts.
- Cancelled/destroyed status still wins over missing worktree.

Strict delivery rules:
- Follow the repo AGENTS.md and .awf/workspace.yml planning contract.
- Write or update focused failing tests first, then implement until green.
- Do not reduce coverage thresholds, workspace coverage requirements, PRD coverage targets, or quality gates.
- Do not add fake tests, broad monkeypatches that skip behavior, xfail/skip shortcuts, or empty assertions.
- Keep changes scoped to the owned paths below unless the plan explains a necessary adjacent edit.
- Run the requested validation commands before finishing.
- Let AWF own push, PR creation, PR monitoring, and merge.

---
Validation: 7 profile command(s) passed inside the workspace container.

### CODE REVIEW

File: src/awf/control/executor.py [hunk 1]
```diff
@@ -1207,6 +1259,72 @@ async def _recover_feature_branch_remote_push_branch(
...
+            if ws.status != expected.value:
+                await self._record_stale_action_skip(
+                    repo,
+                    ws,
+                    action=action,
+                    expected=expected,
+                    reason_code="EXECUTOR_STALE_STATUS",
+                )
+                await session.commit()
+                return False
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/control/executor.py]: fixed in commit bf8aac729b51f58a4bd965e781bcc56f246a38b9

File: src/awf/control/executor.py [hunk 2]
```diff
@@ -314,6 +316,13 @@ async def execute(
         # matches the base branch afterwards, the agent produced zero change
         # and we fail with a specific reason rather than pushing nothing.
         worktree_host = self._config.worktrees_root / workspace_id
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/control/executor.py]: fixed in commit ba190411fd36d57e3a5720ff7d5ec3631f5b4f3c

File: tests/unit/control/test_executor_error_paths.py [hunk 3]
```diff
@@ -484,6 +544,151 @@ async def _load_then_operator_control(workspace_id: str) -> Any:
...
+class TestMissingWorktreeFailure:
+    @pytest.mark.unit
+    async def test_missing_worktree_before_post_agent_commit_marks_infrastructure_failure(
+        self,
+        fake: FakeCommandRunner,
+        factory: async_sessionmaker[AsyncSession],
+        tmp_path: Path,
+    ) -> None:
+        ws_id = await _seed_ready(factory, create_worktree=False)
+        worktree_path = _test_worktree_path(factory, ws_id)
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply tests/unit/control/test_executor_error_paths.py:579]: fixed in commit d43742f7e823db726b5c15e7ddb1369ea99bf68a
